src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -234,6 +239,20 @@ public void init(AdaptorContext context) throws Exception {
       }
     }
 
+    if (!isNullOrEmptyString(cfg.getValue("db.extraMetadataSql"))) {
```

Comment:
This can't be null; we've been drawing the conditions more narrowly in new code. I prefer using foo.isEmpty, but "".equals(foo) is popular in the code, too.